File: backend/rust-actix/src/controllers/healthcheck.rs
use actix_web::{get, HttpResponse, Responder};
use serde::Serialize;

#[derive(Serialize)]
pub struct HealtcheckResponse {
    pub status: String,
    pub message: String,
}

#[get("/healthcheck")]
async fn index() -> impl Responder {
    const MESSAGE: &str = "Simple TODO API with Rust and Actix Web";

    let response_json = &HealtcheckResponse {
        status: "available".to_string(),
        message: MESSAGE.to_string(),
    };

    HttpResponse::Ok().json(response_json)
}

File: backend/golang-fiber/internals/handlers/healthcheck.go
package handlers

import (
	"github.com/gofiber/fiber/v2"
)

// HealthCheck is a simple health check endpoint
func HealthCheck(c *fiber.Ctx) error {
	return c.Status(fiber.StatusOK).JSON(fiber.Map{
		"status":  "available",
		"message": "Simple TODO API with Golang and Fiber",
	})
}
